src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/SimpleDocumentDbRepository.java

```diff
@@ -43,15 +43,23 @@ public SimpleDocumentDbRepository(DocumentDbEntityInformation<T, ID> metadata,
         this.operation = applicationContext.getBean(DocumentDbOperations.class);
         this.information = metadata;
 
-        collection = createCollectionIfNotExists();
+        if (this.information.isAutoCreateCollection()) {
+            collection = createCollectionIfNotExists();
+        } else {
+            collection = null;
```

Comment:
Won;t it result in NRE?
What;s the recommended pattern for consumption?

Reply:
This `collection` field is never used in this class. 
Don't want to remove it in case some customer is relying on this using java.reflection. 
This will only be null if user explicitly uses the annotation of not creating the collection automatically, so they should be aware of this field being null.

Reply:
Reflection based usage is not a contract and okey to change.


---
In reply to: [360990054](https://github.com/microsoft/spring-data-cosmosdb/pull/462#discussion_r360990054) [](ancestors = 360990054)

Reply:
Cool, will remove this field in next iteration.

Reply:
Done.